source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -116,13 +119,24 @@ def move(self, unit, direction, endPoint=None):
 		else:  # moving by a unit other than word
 			res = super(consoleUIATextInfo, self).move(unit, direction,
 														endPoint)
+		if not endPoint:
+			# #10191: IUIAutomationTextRange::move in consoles does not correctly produce a collapsed range
+			# after moving.
+			# Therefore manually collapse.
+			self.collapse()
 		try:
 			if (
-				oldRange
-				and isTextRangeOffscreen(self._rangeObj, visiRanges)
-				and not isTextRangeOffscreen(oldRange, visiRanges)
+				oldInfo
```

Comment:
Could you add a small comment that explains what this if statement tries to do? It is hard to read at first glance.